testsuite/integration/basic/src/test/java/org/wildfly/test/integration/agroal/AgroalDatasourceTestBase.java

```diff
@@ -0,0 +1,295 @@
+        }
+
+        ModelNode subsystemOp = new ModelNode();
+        subsystemOp.get(OP_ADDR).set(SUBSYSTEM, DATASOURCES_SUBSYSTEM);
+        subsystemOp.get(OP).set(ADD);
+
+        try {
+            executeOperation(subsystemOp);
+        } catch (MgmtOperationException e) {
+            log.infof( "Can't add subsystem '%s' by cli: %s", DATASOURCES_SUBSYSTEM, e.getResult().get(FAILURE_DESCRIPTION));
+        }
+
+        // Reload before continuing
+        ServerReload.executeReloadAndWaitForCompletion(getModelControllerClient(), TimeoutUtil.adjust(50000));
+    }
+
+    @After
+    public void disableExtension() throws Exception {
+        ModelNode operation = new ModelNode();
+        operation.get(OP_ADDR).set(new ModelNode().setEmptyList()).add(EXTENSION, AGROAL_EXTENTION);
+        operation.get(OP).set(REMOVE);
+
+        try {
+            executeOperation(operation);
+        } catch (MgmtOperationException e) {
+            log.warnf("Can't remove extension '%s' by cli: ", AGROAL_EXTENTION, e.getResult().get(FAILURE_DESCRIPTION));
+        }
+    }
+
+    // --- //
+
+    @Test
+    public void addDatasource() throws Exception {
+        Datasource ds = Datasource.Builder("testDatasourceEnabled").build();
+
+        try {
+            createDriver(ds);
+            createDataSource(ds);
+            testConnection(ds);
+        } finally {
+            removeDataSourceSilently(ds);
+            removeDriverSilently(ds);
+        }
+    }
+
+    @Test
+    public void allBySystemProperty() throws Exception {
+        String url = "myds.url";
+        String username = "myds.username";
+        String password = "myds.password";
+        String jndiName = "myds.jndi";
+        String driverModule= "myds.driver.module";
+        String driverClass = "myds.driver.class";
+
+        Datasource ds = Datasource.Builder("testAllBySystem")
+                                  .connectionUrl(wrapProp(url))
+                                  .userName(wrapProp(username))
+                                  .password(wrapProp(password))
+                                  .jndiName(wrapProp(jndiName))
+                                  .driverName("h2_ref")
+                                  .driverModule(wrapProp(driverModule))
+                                  .driverClass(wrapProp(driverClass))
+                                  .build();
+
+        try {
+            Datasource defaultPropertyDs = Datasource.Builder("temporary").build();
+            addSystemProperty(url, defaultPropertyDs.getConnectionUrl());
+            addSystemProperty(username, defaultPropertyDs.getUserName());
+            addSystemProperty(password, defaultPropertyDs.getPassword());
+            addSystemProperty(jndiName, defaultPropertyDs.getJndiName());
+            addSystemProperty(driverModule, defaultPropertyDs.getDriverModule());
+            addSystemProperty(driverClass, defaultPropertyDs.getDriverClass());
+            
```

Comment:
The spaces from this line need to be removed as it's causing a checkstyle issue.